aten/src/ATen/native/cuda/fused_adam_impl.cu

```diff
@@ -40,7 +40,15 @@ void _fused_adam_cuda_impl_(
         multi_tensor_apply_for_fused_optimizer<4>(
             tensor_lists,
             state_steps,
-            FusedAdamMathFunctor<scalar_t, 4, ADAM_MODE::ORIGINAL, false>(),
+            FusedAdamMathFunctorMP<
+                scalar_t,
+                ::detail::scalar_type(params[0].scalar_type()),
```

Comment:
i have not figured out how to pass these types in so this code does not even build